### PR TITLE
remove usage of deprecated functions

### LINKED ISSUE #96: Use of deprecated `compare_and_swap` and `spin_loop_hint`
Please replace them
```rust
warning: use of deprecated function `std::sync::atomic::spin_loop_hint`: use hint::spin_loop instead
 --> src/node.rs:2:26
  |
2 | use core::sync::atomic::{spin_loop_hint, AtomicBool, AtomicI64, Ordering};
  |                          ^^^^^^^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: use of deprecated function `std::sync::atomic::spin_loop_hint`: use hint::spin_loop instead
   --> src/node.rs:402:13
    |
402 |             spin_loop_hint();
    |             ^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/map.rs:485:30
    |
485 |             if self.size_ctl.compare_and_swap(sc, -1, Ordering::SeqCst) == sc {
    |                              ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/map.rs:602:22
    |
602 |                     .compare_and_swap(size_ctl, -1, Ordering::SeqCst)
    |                      ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/map.rs:661:22
    |
661 |                     .compare_and_swap(size_ctl, rs + 2, Ordering::SeqCst)
    |                      ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/map.rs:730:22
    |
730 |                     .compare_and_swap(next_index, next_bound, Ordering::SeqCst)
    |                      ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/map.rs:780:34
    |
780 |                 if self.size_ctl.compare_and_swap(sc, sc - 1, Ordering::SeqCst) == sc {
    |                                  ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
    --> src/map.rs:1148:30
     |
1148 |             if self.size_ctl.compare_and_swap(sc, sc + 1, Ordering::SeqCst) == sc {
     |                              ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
    --> src/map.rs:1213:34
     |
1213 |                 if self.size_ctl.compare_and_swap(sc, sc + 1, Ordering::SeqCst) == sc {
     |                                  ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicIsize::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
    --> src/map.rs:1216:37
     |
1216 |             } else if self.size_ctl.compare_and_swap(sc, rs + 2, Ordering::SeqCst) == sc {
     |                                     ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicI64::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/node.rs:334:14
    |
334 |             .compare_and_swap(0, WRITER, Ordering::SeqCst)
    |              ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicI64::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/node.rs:357:22
    |
357 |                     .compare_and_swap(state, WRITER, Ordering::SeqCst)
    |                      ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicI64::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/node.rs:391:22
    |
391 |                     .compare_and_swap(state, state | WAITER, Ordering::SeqCst)
    |                      ^^^^^^^^^^^^^^^^

warning: use of deprecated associated function `std::sync::atomic::AtomicI64::compare_and_swap`: Use `compare_exchange` or `compare_exchange_weak` instead
   --> src/node.rs:459:18
    |
459 |                 .compare_and_swap(s, s + READER, Ordering::SeqCst)
    |                  ^^^^^^^^^^^^^^^^

warning: 14 warnings emitted

```

### PR DESCRIPTION
**Pull request by qiujiangkun**
closes #96 

This change is [Reviewable](https://reviewable.io/reviews/jonhoo/flurry/97)

### CODE REVIEW

File: src/map.rs [hunk 1]
```diff
@@ -195,8 +195,8 @@ pub struct TryInsertError<'a, V> {
 }
 
 impl<'a, V> Display for TryInsertError<'a, V>
-where
-    V: Debug,
+    where
```
Comment (jonhoo) [inline src/map.rs:198]: These changes don't seem right — have you run your code through `rustfmt`? Is your version perhaps old somehow? I get no differences if I run `cargo fmt` locally.

File: src/map.rs [hunk 2]
```diff
@@ -482,7 +482,7 @@ impl<K, V, S> HashMap<K, V, S> {
                 continue;
             }
 
-            if self.size_ctl.compare_and_swap(sc, -1, Ordering::SeqCst) == sc {
+            if self.size_ctl.compare_exchange(sc, -1, Ordering::SeqCst, Ordering::Relaxed) == Ok(sc) {
```
Comment (jonhoo) [inline src/map.rs:485]: This works, but is overly complex. With `compare_exchange` you should be able to use `.is_ok()` instead.

**Comment by qiujiangkun** [comment]
Can we use `Ordering::AcqRel` instead of `Ordering::SeqCst`? It seems that the former has better performance.

File: src/map.rs [hunk 3]
```diff
@@ -599,8 +603,8 @@ where
                 // try to aquire the initialization "lock" to indicate that we are initializing the table.
                 if self
                     .size_ctl
-                    .compare_and_swap(size_ctl, -1, Ordering::SeqCst)
-                    != size_ctl
+                    .compare_exchange(size_ctl, -1, Ordering::SeqCst, Ordering::Relaxed)
```
Comment (jonhoo) [inline src/map.rs:606]: This should be `.is_err()`.

File: src/map.rs [hunk 4]
```diff
@@ -2919,6 +2939,7 @@ where
         head
     }
 }
+
```
Comment (jonhoo) [inline src/map.rs:2942]: This seems like an unrelated change?

File: src/map.rs [hunk 5]
```diff
@@ -3117,6 +3138,7 @@ fn capacity() {
 #[cfg(test)]
 mod tests {
     use super::*;
+
```
Comment (jonhoo) [inline src/map.rs:3141]: This and the other new newlines below also seem like unrelated changes?

**Review by jonhoo — CHANGES_REQUESTED** [review]
Ah, this looks much better, thanks! Just a few small nits left.

Memory ordering is [complicated](https://www.youtube.com/watch?v=rMGWeSjctlY), so it's not quite a matter of "just changing it" — it _might_ be safe to change some of them, but it would require careful analysis for each case.

File: src/map.rs (hunk 5 above)
Comment (qiujiangkun) [reply src/map.rs]: It seems like my IDE formatter added and then cargo fmt decides to keep it.

**Review by jonhoo — APPROVED** [review]
Thanks! I'll tidy up the CI errors.
